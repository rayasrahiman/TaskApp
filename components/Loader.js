import {View, ActivityIndicator, StyleSheet} from 'react-native';
import React from 'react';
import {heightPercentageToDP as hp} from 'react-native-responsive-screen';

import {Colors} from '../constants/colors';

export default function Loader() {
  return (
    <View style={styles.container}>
      <ActivityIndicator color={Colors.primary700} size={hp(4)} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
