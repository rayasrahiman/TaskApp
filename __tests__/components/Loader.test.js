import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {shallow} from 'enzyme';

import Loader from '../../components/Loader';

beforeEach(jest.spyOn(console, 'error').mockImplementation(() => undefined));
afterEach(() => jest.clearAllMocks());

describe('Loader', () => {
  test('should render Loader correctly', () => {
    const wrapper = render(<Loader />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
