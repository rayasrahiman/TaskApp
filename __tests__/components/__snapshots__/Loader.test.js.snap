// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render Loader correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#0570c9"
    size={53.5}
  />
</View>
`;
